WooCommerce/UITestsFoundation/XCTest+Extensions.swift

```diff
@@ -114,8 +114,44 @@ extension XCUIElement {
         startCoordinate.press(forDuration: 0.01, thenDragTo: destination)
     }
 
-    public func getTextVisibilityCount(text: String) throws -> Int {
-        let predicate = NSPredicate(format: "label CONTAINS[c] %@", text)
+    func getStaticTextVisibilityCount(textToFind: String) throws -> Int {
+        let predicate = NSPredicate(format: "label CONTAINS[c] %@", textToFind)
         return XCUIApplication().staticTexts.containing(predicate).count
     }
+
+    public func assertTextVisibilityCount(textToFind: String) {
+        let count = try! getStaticTextVisibilityCount(textToFind: textToFind)
+        XCTAssertTrue(count == 1, "Expected '\(textToFind)' to appear once, but it appeared '\(count)' times!")
+    }
+
+    // Parent is using accessibilityIdentifier, child is using staticText
+    func verifyChildElementOnParentCell(parent: String, child: String) throws -> Bool {
+        let parentPredicate = NSPredicate(format: "identifier == %@", parent)
+        let childPredicate = NSPredicate(format: "label ==[c] %@", child)
+
+        return XCUIApplication().tables.cells.matching(parentPredicate).children(matching: .staticText).element(matching: childPredicate).firstMatch.exists
+    }
+
+    public func assertElementExistsOnCell(mainCell: String, elementToFind: String) {
+        XCTAssertTrue(try verifyChildElementOnParentCell(parent: mainCell, child: elementToFind), "Element does not exist on cell!")
+    }
+
+    func verifyLabelContainsTwoPredicates(firstSubstring: String, secondSubstring: String) throws -> Bool {
+        let firstPredicate = NSPredicate(format: "label CONTAINS[c] %@", firstSubstring)
+        let secondPredicate = NSPredicate(format: "label CONTAINS[c] %@", secondSubstring)
+        let predicateCompound = NSCompoundPredicate.init(type: .and, subpredicates: [firstPredicate, secondPredicate])
+
+        return XCUIApplication().staticTexts.containing(predicateCompound).count == 1
+    }
+
+    public func assertTwoTextsAppearOnSameLabel(firstSubstring: String, secondSubstring: String) {
+        XCTAssertTrue(try verifyLabelContainsTwoPredicates(firstSubstring: firstSubstring, secondSubstring: secondSubstring),
+        """
+        '\(firstSubstring)' and '\(secondSubstring)' does not appear on the same label!
+        """)
+    }
+
+    public func assertCorrectCellCountDisplayed(expectedCount: Int, actualCount: Int) {
+        XCTAssertEqual(expectedCount, actualCount, "Expecting '\(expectedCount)' reviews, got '\(actualCount)' instead!")
+    }
```

Comment:
I'm not sure this encapsulation of `XCTAssertEqual` with a custom failure message is useful for the code clarity 🤔 

How do you feel about leaving the `XCTAssertEqual` call inline? Also, unless I'm mistaken, the usage of this [in `ProductsScreen`](https://github.com/woocommerce/woocommerce-ios/pull/5776/files#diff-5190a256a7ecbacd3c33603133dec2be6085d203edc8798ddff5c9625522f0b8R56), shouldn't fail with an error saying "reviews", because the screen actually shows products. 

Let me know if I'm missing something about this method 😅

Reply:
This is used on both the [ProductsScreen](https://github.com/woocommerce/woocommerce-ios/blob/7e02413b6ad7b0b73f9011d9ca597a1614086b61/WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift#L56) and [ReviewsScreen](https://github.com/woocommerce/woocommerce-ios/blob/7e02413b6ad7b0b73f9011d9ca597a1614086b61/WooCommerce/UITestsFoundation/Screens/Reviews/ReviewsScreen.swift#L43). And yeah the error message is incorrect, it says `reviews` instead of `products` because that was the last one I worked on 😬 , will update that to be more generic.

The reason that I moved this here instead of inline is so that on [verifyProductListOnProductsScreen](https://github.com/woocommerce/woocommerce-ios/blob/7e02413b6ad7b0b73f9011d9ca597a1614086b61/WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift#L53-L59) all the assertions will be using `app.assertSomething()` and not a combination of two assert types like
```
func verifyScreen {
    app.assertSomething()
    XCTAssertEqual(assertSomething)
}
```